core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -486,18 +487,24 @@ private StartupConsistencyChecker(Long fileId) {
             inconsistentUris.add(parentUri);
           }
           for (Inode childInode : ((InodeDirectory) parentInode).getChildren()) {
-            AlluxioURI childUri = parentUri.join(childInode.getName());
-            if (childInode.isDirectory()) {
-              dirsToCheck.add(childInode.getId());
-            } else {
-              childInode.lockRead();
-              try {
+            try {
+              childInode.lockReadAndCheckParent(parentInode);
```

Comment:
I've created this ticket to track the issue: https://alluxio.atlassian.net/browse/ALLUXIO-2458